### PR TITLE
e2e: improvements

### PR DESCRIPTION
**Pull request by invidian**
To simplify changes in #165.

### CODE REVIEW

File: .github/workflows/push_pr.yaml [hunk 1]
```diff
@@ -50,3 +50,16 @@ jobs:
...
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    steps:
+      - uses: actions/checkout@v2
+      - name: Codespell test
+        uses: codespell-project/actions-codespell@master
+        with:
+          skip: ./.git
+          check_filenames: true
+          check_hidden: true
```
Comment (roobre) [inline .github/workflows/push_pr.yaml:65]: Sneaky change :P

File: e2e/charts/newrelic-infrastructure-k8s-e2e/templates/daemonset.yaml [hunk 2]
```diff
@@ -54,12 +54,10 @@ spec:
             {{- end }}
           env:
             {{- if eq .Values.daemonset.clusterFlavor "Minikube" }}
-              {{- if and (ne .Values.daemonset.clusterVersion "v1.15.x") (ne .Values.daemonset.clusterVersion "v1.14.x") (ne .Values.daemonset.clusterVersion "v1.13.x") }}
+              {{- if semverCompare ">=1.16-0" .Capabilities.KubeVersion.GitVersion }}
```
Comment (roobre) [inline e2e/charts/newrelic-infrastructure-k8s-e2e/templates/daemonset.yaml:57]: What does the `-0` do?

File: e2e/charts/newrelic-infrastructure-k8s-e2e/templates/daemonset.yaml [hunk 3]
```diff
@@ -54,12 +54,10 @@ spec:
             {{- end }}
           env:
             {{- if eq .Values.daemonset.clusterFlavor "Minikube" }}
```
Comment (roobre) [inline e2e/charts/newrelic-infrastructure-k8s-e2e/templates/daemonset.yaml:56]: I'd take the advantage of your enlightment about what these checks do and add a small comment describing them.

File: e2e/cmd/e2e.go [hunk 4]
```diff
@@ -287,117 +275,106 @@ func waitForKSM(c *k8s.Client, logger *logrus.Logger) (*v1.Pod, error) {
...
 	if err != nil {
 		execErr.errs = append(execErr.errs, err)
 	}
 
-	if currentScenario.ClusterFlavor == minikubeFlavor {
+	if se.scenario.ClusterFlavor == minikubeFlavor {
 		// Minikube use hostPath provisioner for PVCs, which makes kubelet to not report PVC volumes in /stats/summary
 		// See https://github.com/yashbhutwala/kubectl-df-pv/issues/2 for more info.
-		logger.Info("Skipping `testSpecificEntities` because you're running them in Minikube.")
+		se.logger.Info("Skipping `testSpecificEntities` because you're running them in Minikube.")
```
Comment (roobre) [inline e2e/cmd/e2e.go:336]: Again not quite a fan of the `testSpecificEntities` name, I think we should reflect what the comment above says in the log line (and then possibly remove the comment).

File: e2e/cmd/e2e.go [hunk 5]
```diff
@@ -128,19 +86,16 @@ const (
 
 var allJobs = [...]job{jobKSM, jobKubelet, jobScheduler, jobEtcd, jobControllerManager, jobAPIServer}
 
-func execIntegration(pod v1.Pod, ksmPod *v1.Pod, dataChannel chan integrationData, wg *sync.WaitGroup, c *k8s.Client, logger *logrus.Logger) {
-	defer timer.Track(time.Now(), fmt.Sprintf("execIntegration func for pod %s", pod.Name), logger)
-	defer wg.Done()
```
Comment (roobre) [inline e2e/cmd/e2e.go:133]: Here we got rid of a waitgroup and a channel. What were these doing?

File: e2e/charts/newrelic-infrastructure-k8s-e2e/templates/daemonset.yaml (hunk 2 above)
Comment (invidian) [reply e2e/charts/newrelic-infrastructure-k8s-e2e/templates/daemonset.yaml:57]: https://github.com/Masterminds/semver#working-with-prerelease-versions IIRC it includes the prereleases.

File: e2e/charts/newrelic-infrastructure-k8s-e2e/templates/daemonset.yaml (hunk 3 above)
Comment (invidian) [reply e2e/charts/newrelic-infrastructure-k8s-e2e/templates/daemonset.yaml:56]: Sure, added some.

File: e2e/cmd/e2e.go (hunk 4 above)
Comment (invidian) [reply e2e/cmd/e2e.go:336]: Sure, but perhaps out of scope for this already large PR?

File: e2e/cmd/e2e.go (hunk 5 above)
Comment (invidian) [reply e2e/cmd/e2e.go:133]: So, integration was executed in parallel on all listed pods. Commit message explains that IIRC.

But as integration runs fast and usually on one pod only, there is little value it seems in executing it asynchronously.

File: e2e/cmd/e2e.go (hunk 4 above)
Comment (roobre) [reply e2e/cmd/e2e.go:336]: Yup, you're right. This is already pretty big.

**Review by roobre — APPROVED** [review]
Looking great, thanks a lot @invidian

**Pull request merged**